net-irc/znc/znc-9999.ebuild

```diff
@@ -2,163 +2,170 @@
 
 RDEPEND="
-	dev-libs/icu:=
-	sys-libs/zlib
-	perl? ( >=dev-lang/perl-5.10 )
+	icu? ( dev-libs/icu:= )
+	nls? ( dev-libs/boost[nls] )
+	perl? ( >=dev-lang/perl-5.10:= )
 	python? ( ${PYTHON_DEPS} )
 	sasl? ( >=dev-libs/cyrus-sasl-2 )
 	ssl? (
-		!libressl? ( dev-libs/openssl:0 )
-		libressl? ( dev-libs/libressl )
+		!libressl? ( dev-libs/openssl:0= )
+		libressl? ( dev-libs/libressl:0= )
 	)
 	tcl? ( dev-lang/tcl:0= )
+	zlib? ( sys-libs/zlib )
 "
-DEPEND="${RDEPEND}
+DEPEND="
+	${RDEPEND}
+	${SWIG_DEPEND}
+	nls? ( sys-devel/gettext )
 	virtual/pkgconfig
-	perl? (
-		>=dev-lang/swig-2.0.12
-	)
-	python? (
-		>=dev-lang/swig-2.0.12
-	)
 "
 
-S=${WORKDIR}/${PN}-${MY_PV}
-
-CONFDIR="/var/lib/znc"
-
 pkg_setup() {
 	if use python; then
 		python-single-r1_pkg_setup
 	fi
 	if use daemon; then
 		enewgroup ${PN}
-		enewuser ${PN} -1 -1 /dev/null ${PN}
+		enewuser ${PN} -1 -1 /var/lib/${PN} ${PN}
+		# The home directory was previously set to /dev/null
+		# This caused a bug with the systemd unit
+		# https://bugs.gentoo.org/521916
+		esethome ${PN} /var/lib/${PN}
 	fi
 }
 
 src_prepare() {
-	AT_M4DIR="${S}/m4" \
-		eautoreconf
-
-	# build system quirk, it does not define AM_INIT_AUTOMAKE, nor
-	# does it have Makefile.am in the root dir, but we need '--add-missing'
-	automake --add-missing
+	l10n_find_plocales_changes "${S}/src/po" "${PN}." '.po'
+	remove_locale() {
+		# Some languages can miss some modules
+		rm src/po/${PN}.${1}.po modules/po/*.${1}.po || true
+	}
+	l10n_for_each_disabled_locale_do remove_locale
+	cmake-utils_src_prepare
 }
 
 src_configure() {
-	econf \
-		$(use_enable debug) \
-		$(use_enable ipv6) \
-		$(use_enable perl) \
-		$(use python && echo "--enable-python=python3") \
-		$(use_enable sasl cyrus) \
-		$(use_enable ssl openssl) \
-		$(use_enable tcl tcl)
+	local want_swig
+	if [[ ${PV} == *9999* ]]; then
+		want_swig=yes
+	else
+		want_swig=no
+	fi
+	local mycmakeargs=(
+		-DWANT_SYSTEMD="$(usex daemon)"
+		-DSYSTEMD_DIR="$(systemd_get_systemunitdir)"
+		-DWANT_ICU="$(usex icu)"
+		-DWANT_IPV6="$(usex ipv6)"
+		-DWANT_I18N="$(usex nls)"
+		-DWANT_PERL="$(usex perl)"
+		-DWANT_PYTHON="$(usex python)"
+		-DWANT_CYRUS="$(usex sasl)"
+		-DWANT_OPENSSL="$(usex ssl)"
+		-DWANT_TCL="$(usex tcl)"
+		-DWANT_ZLIB="$(usex zlib)"
+		-DWANT_SWIG="${want_swig}"
+	)
+	if use test; then
+		export GTEST_ROOT="${WORKDIR}/googletest-release-${GTEST_VER}/googletest"
+		export GMOCK_ROOT="${WORKDIR}/googletest-release-${GTEST_VER}/googlemock"
+	fi
+	cmake-utils_src_configure
+}
+
+src_test() {
+	pushd "${BUILD_DIR}" > /dev/null || die
+	${CMAKE_MAKEFILE_GENERATOR} unittest || die "Unit test failed"
+	popd > /dev/null || die
 }
 
 src_install() {
-	emake install DESTDIR="${D}"
-	dodoc NOTICE README.md
+	cmake-utils_src_install
+	dodoc NOTICE
 	if use daemon; then
-		newinitd "${FILESDIR}"/znc.initd-r1 znc
+		newinitd "${FILESDIR}"/znc.initd-r2 znc
 		newconfd "${FILESDIR}"/znc.confd-r1 znc
 	fi
+	DOC_CONTENTS=$(<"${FILESDIR}/README.gentoo") || die
```

Comment:
This one is at least outdated now that we are no longer shipping our `--system-wide-config-as` parameter.